adr/0007-placement-support.md

```diff
@@ -0,0 +1,120 @@
+- the assembly of the solution before the engine is run
+- the focus of a template run on a single segment
+- the generation of account templates uses different concepts to those of product templates
+- the maintenance of critical configuration information such as environment to account mappings external to the cmdb.
+
+The introduction of dynamic cmdb loading and the input pipeline processor with full cmdb qualification presents an opportunity to address these long standing issues and complete the implementation of placement support. The question is how.
+
+## Decision Drivers <!-- optional -->
+
+* reuse existing concepts and ways of doing things as much as possible
+* given a cmdb, no other configuration information should be necessary to deploy a product
+
+## Considered Options
+
+This ADR extends on the decision made in [ADR-0002](../0002-layer-deployments.md) - Layer Level Deployments and provides more design detail, primarily by the enrichment of link semantics.
+
+## Decision Outcome
+
+Chosen option: "Extend link semantics", because it addresses the decision drivers best (see below) and aligns with the decision previously made.
+
+### Positive Consequences <!-- optional -->
+
+* reduced configuration complexity - everything in one place
+* better validation of configurations
+* links have proven very flexible so expect the porposed changes to be repurposed for other things in the future
+* straightforward transition process once infrastructure in place
+
+### Negative Consequences <!-- optional -->
+
+* Expansion of semantics of existing concepts may be harder to document, understand and explain than dedicated new concepts
+
+## Design Detail
+
+### New Link Semantics
+
+Currently links are used extensively in product solutions. A link goes from an occurrence in one solution to an occurrence in another (possibly same) solution. This option builds on existing link processing as follows;
+
+1. A link targets an occurrence within a solution.
+1. A link uses attributes to identify a solution and the specific occurrence within that solution.
+1. Occurrences within a solution are identified by a standard set of link attributes
+     - `Tier`
+     - `Component`
+     - `Subcomponent`
+     - `Instance`
+     - `Version`
+
+   Thus these attributes must be present on a link, or be inherited from the occurrence on which the link is configured.
+1. The `Subcomponent` attribute replaces use of type specific attributes such as `Port`.
+1. An optional `Type` attribute is added to ensure links to same named subcomponents can be resolved. Its value is the component type of the link target.
+1. Each solution has a scope and an identifier within the scope.
+1. A link must have a `Scope` attribute representing the target-solution's scope. The default, representing existing usage, is the `segment` scope.
```

Comment:
The default would actually be self rather than segment right? 
A link configured to be scope segment could be another segment 

So I think here the default would a link within my solution

Reply:
I've updated the text to reflect latest thinking on `districts`. Suggest a reread/review.